docs/cancel-and-close-unpaid-orders-after-two-days/script.liquid

```diff
@@ -56,143 +28,254 @@
-    {% endunless %}
+  {% if event.preview %}
+    {% capture result_json %}
+      {
+        "data": {
+          "orders": {
+            "edges": [
+              {
+                "node": {
+                  "id": "gid://shopify/Order/1234567890",
+                  "name": "#1234",
+                  "processedAt": "1990-01-01",
+                  "cancelledAt": null,
+                  "closedAt": null,
+                  "tags": [
+                    {{ options.only_process_orders_having_this_tag | strip | json }}
+                  ],
+                  "suggestedRefund": {
+                    "refundLineItems": [
+                      {
+                        "lineItem": {
+                          "id": "gid://shopify/LineItem/1234567890",
+                          "fulfillableQuantity": 5,
+                          "restockable": true
+                        },
+                        "location": {
+                          "id": "gid://shopify/Location/1234567890"
+                        }
+                      }
+                    ]
+                  },
+                  "transactions": [
+                    {
+                      "id": "gid://shopify/OrderTransaction/1234567890",
+                      "gateway": "bogus"
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        }
+      }
+    {% endcapture %}
+
+    {% assign result = result_json | parse_json %}
   {% endif %}
 
-  {% unless summary.qualifies_for_cancellation or summary.qualifies_for_closing or summary.qualifies_for_tagging %}
-    {% continue %}
-  {% endunless %}
+  {% for edge in result.data.orders.edges %}
+    {% assign order = edge.node %}
 
-  {% if options.test_mode__boolean %}
-    {% assign test_mode_summaries[test_mode_summaries.size] = summary %}
-    {% continue %}
-  {% endif %}
+    {% assign processed_at_s = order.processedAt | default: order.createdAt | date: "%s" | times: 1 %}
+
+    {% if processed_at_s > maximum_date_s %}
+      {% continue %}
+    {% endif %}
+
+    {% assign summary = hash %}
+    {% assign summary["order_name"] = order.name %}
+    {% assign summary["order_id"] = order.id %}
+    {% assign summary["order_processed_at"] = processed_at_s | date: "%FT%T%:z" %}
+    {% assign summary["threshold_processed_at"] = maximum_date_s | date: "%FT%T%:z" %}
+
+    {% if order.cancelledAt == blank %}
+      {% assign summary["qualifies_for_cancellation"] = true %}
+    {% else %}
+      {% assign summary["qualifies_for_cancellation"] = false %}
+    {% endif %}
+
+    {% if order.closedAt == blank %}
+      {% assign summary["qualifies_for_closing"] = true %}
+    {% else %}
+      {% assign summary["qualifies_for_closing"] = false %}
+    {% endif %}
+
+    {% assign summary["qualifies_for_tagging"] = false %}
+    {% if options.tag_to_add_to_the_order != blank %}
+      {% assign order_tags = order.tags | downcase %}
```

Comment:
Missed this earlier - `order.tags` is an array, yeah? Downcasing that results in a string, which means that using `contains` on it is going to return false positives, because it's just a substring check at that point.

Reply:
Ahh, it looks like an array on echo.... but that entire array is a string, got it.